novawallet/Modules/NetworksList/NetworksListInteractor.swift

```diff
@@ -0,0 +1,50 @@
+import UIKit
+import RobinHood
+import SubstrateSdk
+
+final class NetworksListInteractor {
+    weak var presenter: NetworksListInteractorOutputProtocol?
+
+    let chainRegistry: ChainRegistryProtocol
+
+    private var chains: [ChainModel.Id: ChainModel] = [:]
+
+    init(chainRegistry: ChainRegistryProtocol) {
+        self.chainRegistry = chainRegistry
+    }
+
+    private func subscribeChains() {
+        chainRegistry.chainsSubscribe(
+            self,
+            runningInQueue: .main
+        ) { [weak self] changes in
+            guard let self else { return }
+            presenter?.didReceiveChains(changes: changes)
+
+            chains = changes.mergeToDict(chains)
```

Comment:
I don't think we need to accumulate chains here. I would rather subscribe to the chain state once it is new and unsubscribe when delete